phoenix-core/src/main/java/org/apache/phoenix/mapreduce/VerifyReplicationTool.java

```diff
@@ -0,0 +1,477 @@
+                        } else if (!sourceValues.equals(targetValues)) {
+                            valuesMatch = false;
+                        }
+                        if (!valuesMatch) {
+                            logFailRowAndIncrementCounter(context, Counter.CONTENT_DIFFERENT_ROWS, value.getSourceKey());
+                        } else {
+                            context.getCounter(Counter.GOODROWS).increment(1);
+                        }
+                        targetHasNext = targetResultSet.next();
+                        break;
+                    } else if (keyCompare < 0) {
+                        // row only exists in source table
+                        logFailRowAndIncrementCounter(context, Counter.ONLY_IN_SOURCE_TABLE_ROWS, value.getSourceKey());
+                        break;
+                    } else {
+                        // row only exists in target table
+                        logFailRowAndIncrementCounter(context, Counter.ONLY_IN_TARGET_TABLE_ROWS,
+                                targetKey);
+                        targetHasNext = targetResultSet.next();
+                    }
+                }
+            } catch (SQLException e) {
+                throw new IOException(e.getMessage());
+            }
+        }
+
+        @Override
+        protected void cleanup(Context context) throws IOException, InterruptedException {
+            super.cleanup(context);
+            try {
+                if (!sourceHasData && targetResultSet == null) {
+                    getTargetResultSet(context.getConfiguration(), null, null);
+                }
+                if(targetResultSet != null) {
+                    while (targetHasNext) {
+                        ImmutableBytesWritable targetKey = null;
+
+                        if (targetResultSet.getCurrentRow() != null) {
+                            targetKey = new ImmutableBytesWritable();
+                            targetResultSet.getCurrentRow().getKey(targetKey);
+                        }
+                        logFailRowAndIncrementCounter(context, Counter.ONLY_IN_TARGET_TABLE_ROWS,
+                                targetKey);
+                        targetHasNext = targetResultSet.next();
+                    }
+                }
+            } catch (SQLException e) {
+                LOG.error("Unable to properly scan the peer table", e);
+            } finally {
+                try {
+                    targetResultSet.close();
+                    targetResultSet = null;
+                } catch (SQLException e) {
+                    LOG.error("Unable to close peer result set", e);
+                }
+            }
+
+        }
+
+        private void getTargetResultSet(Configuration conf, byte[] startRow, byte[] stopRow) throws IOException, SQLException {
+            Scan targetScan = targetQueryPlan.getContext().getScan();
+            targetScan.setAttribute(BaseScannerRegionObserver.SKIP_REGION_BOUNDARY_CHECK,
```

Comment:
You don't need this since we skip this check in MR jobs anyways.

Reply:
Oh yeah, that's right. Thanks for pointing it out.